api/test/e2enew/route/route_service_upstream_test.go

```diff
@@ -0,0 +1,277 @@
+			ExpectStatus: http.StatusBadRequest,
+		}),
+		table.Entry("hit invalid route on data plane", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello_",
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        base.SleepTime,
+		}),
+	)
+})
+
+var _ = ginkgo.Describe("route create service", func() {
+	table.DescribeTable("test route create service",
+		func(tc base.HttpTestCase) {
+			base.RunTestCase(tc)
+		},
+		table.Entry("make sure the route has not created", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/server_port",
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        base.SleepTime,
+		}),
+		table.Entry("create service", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/services/200",
+			Body: `{
+				"upstream": {
+				"type": "roundrobin",
+				"nodes": [
+						{
+							"host": "` + base.UpstreamIp + `",
+							"port": 1980,
+							"weight": 1
+						},
+						{
+							"host": "` + base.UpstreamIp + `",
+							"port": 1981,
+							"weight": 1
+						},
+						{
+							"host": "` + base.UpstreamIp + `",
+							"port": 1982,
+							"weight": 1
+						}
+					]
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("create route using the service just created", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"name": "route1",
+				"uri": "/server_port",
+				"service_id": "200"
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			Sleep:        base.SleepTime,
+		}),
+	)
+	ginkgo.It("batch test /server_port api", func() {
+		// sleep for etcd sync
+		time.Sleep(time.Duration(300) * time.Millisecond)
+
+		// batch test /server_port api
+		res := base.BatchTestServerPort(18)
+
+		gomega.Expect(res["1980"]).Should(gomega.Equal(6))
+		gomega.Expect(res["1981"]).Should(gomega.Equal(6))
+		gomega.Expect(res["1982"]).Should(gomega.Equal(6))
+	})
+	table.DescribeTable("delete route and service",
+		func(tc base.HttpTestCase) {
+			base.RunTestCase(tc)
+		},
+		table.Entry("delete route", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("remove service", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/services/200",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("hit deleted route", base.HttpTestCase{
```

Comment:
Bad title, how can we hit a deleted route?

Reply:
done.